src/cosmos-keys.ts

```diff
@@ -49,16 +49,21 @@ export function getSeed(randomBytesFunc: (size: number) => Buffer = randomBytes)
   return mnemonic
 }
 
-export function getNewWallet(randomBytesFunc: (size: number) => Buffer = randomBytes): Wallet {
+export function getNewWallet(randomBytesFunc: (size: number) => Buffer = randomBytes, networkPrefix: string): Wallet {
   const mnemonic = getSeed(randomBytesFunc)
-  return getNewWalletFromSeed(mnemonic)
+  return getNewWalletFromSeed(mnemonic, networkPrefix)
 }
 
 // NOTE: this only works with a compressed public key (33 bytes)
-export function getCosmosAddress(publicKey: Buffer): string {
+export function getCosmosAddress(publicKey: Buffer, networkPrefix: string): string {
   const message = CryptoJS.enc.Hex.parse(publicKey.toString('hex'))
   const address = CryptoJS.RIPEMD160(CryptoJS.SHA256(message) as any).toString()
-  const cosmosAddress = bech32ify(address, `cosmos`)
+  if (networkPrefix === `regen`) {
+    networkPrefix = `xrn:`
+  } else if (networkPrefix === `livepeer`) {
```

Comment:
Nice idea. But I would propose that the integrating application handles this case to not pollute this code base

Reply:
OK, will refactor it

Reply:
Cool! You were fast